### PR TITLE
hugo: Update contribution section of README

### PR DESCRIPTION
**Pull request by moorereason**
Add an explicit commit message guidelines section to the README.  Keep
all of the guidelines from Chris Beams except for the 72 character line
limit (we don't follow that, nor does the Go team).  Add three new
guidelines: package prefix in subject, references in body, and
encouragement of message body in general.

Add a new section on using Git Remotes.

### CODE REVIEW

**Comment by digitalcraftsman** [comment]
Good idea to add examples for clarification. Be aware that we also have a `CONTRIBUTING.md` that contains the same information.

File: README.md [hunk 1]
```diff
@@ -105,20 +105,97 @@ The Hugo community and maintainers are very active and helpful and the project b
...
 
+### Git Commit Message Guidelines
+
+Quality Git commit messages are critical to a large project;
+therefore, we've established the following guidelines:
+
+1. Prefix the subject with the primary affected **package**.
+1. After the package prefix, **capitalize** the subject.
+1. End the subject **without punctuation**.
+1. Use the **imperative** mood in the subject and body.
```
Comment (bep) [inline README.md]: No sure what Chris Beams says a bout the body and mood, but this will be impossible to enforce.

**Comment by bep** [comment]
While I agree about the content, I fear that this will make people hesitate to contribute, it looks like we are very strict. The current text could probably have some clearer examples ... When do merges, I only refer to the commit message guidelines if:

1. It is obvious that the contributor has some experience in development
1. Is a recurring contributor

If no, I just do quick corrections of their commit messages with `git --amend`. Most people appreciate a clean commit log, have to be shown one to see the light ...

File: README.md (hunk 1 above)
Comment (moorereason) [inline README.md]: I'll remove "and body".

**Comment by moorereason** [comment]
Well, I do want us to be more strict in the future, but I don't want to scare off contributors either.  As long as we amend the commit message before committing, I'm happy.  :smile:

**Comment by moorereason** [comment]
I pushed another commit to the PR.  Let me know what you think.

Not sure what to do about `CONTRIBUTING.md`.  Can we just delete that file or should we move the contributing sections from the README into that file?

**Comment by spf13** [comment]
@moorereason I would add a preface to the guidelines. Something like...

The Hugo project welcomes all contributors and contributions regardless of skill or experience level. If you are interested in helping with the project we will help you with your contribution. Hugo is a very active project with many contributions happening daily. Because we want to create the best possible product for our users and the best contribution experience for our developers we have a set of guidelines which ensure that all contributions are acceptable. The guidelines are not intended as a filter or barrier to participation. If you are unfamiliar with the contribution process the Hugo team will help you and teach you how to bring your contribution in accordance with the guidelines. 

Feel free to wordsmith. 
 
My vote is that the readme should have a quick summary of our view on contributors and the process/tools available and then link to a more complete guide in the CONTRIBUTING.md document.

**Comment by moorereason** [comment]
I've made a lot of changes, rebased, and pushed.  See commit message.

**Comment by anthonyfok** [comment]
Hi Cameron,

> 1. Prefix the subject with the primary affected **package**.

I wonder if the term "**package**" in "package prefix" may be a little bit confusing to newcomers, me included.  :grin:  By observing the package prefixes that you have used, I derived that the "package prefix" seems to be the subdirectory under the root directory of the hugo source code.  Is that usually the case?  And is that always the case?

Hence, if you don't mind, could you please consider adding a brief explanation of the term "package" in this context?  Many thanks!  :-)

**Comment by moorereason** [comment]
@anthonyfok, yes, it is the Go package.  I adopted the practice from watching the Go team.  We don't have any sub-packages at this point, so it's generally the subdirectory inside the hugo source code.  For the Go team, you would submit patches with `html/template: Fix something...`.

Does the following change help?

> 1. Prefix the subject with the primary affected **Go package** inside of hugo, followed by a colon.

If that's not enough, what about the following?

> 1. Prefix the subject with the primary affected **Go package** inside of hugo, followed by a colon.  For example, if you submit a change to `github.com/spf13/hugo/tpl`, prefix the subject with `tpl:`.  If it's a change to the documentation, use `docs:` as the prefix.

I was trying to be as concise as possible for the bullet points.

**Comment by digitalcraftsman** [comment]
Sometimes pull requests don't contain any documentation. Mostly it's obvious to mention in a few lines (if neccessary) about a new feature describing how it should be used.

Does it make sense to add another bullet point that reminds contributors about it?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,23 +1,144 @@
...
 
-We welcome your contributions. 
-To make the process as seamless as possible, we ask for the following:
+## Reporting Issues
 
-* Go ahead and fork the project and make your changes. We encourage pull requests to discuss code changes.
+If you believe you have found a defect in Hugo or its documentation, use
+the Github issue tracker to report the problem to the Hugo maintainers.
+If you're not sure if it's a bug or not, start by asking in the [discussion forum](http://discuss.gohugo.io).
+When reporting the issue, please provide the version of Hugo in use (`hugo version`).
```
Comment (spf13) [inline CONTRIBUTING.md]: and your operating system.

**Comment by moorereason** [comment]
@spf13,
I've added another commit to this PR to address your comment and added a line about adding documentation when necessary (thanks @digitalcraftsman).

I'll squash this commit later, but I wanted to make it easy to see the changes.

**Comment by digitalcraftsman** [comment]
I think it would make sense to link to my "How to contribute to Hugo" tutorial (see #2044). This tutorial guides new Git / GitHub users in the right direction and finally links back to the expected structure of an issue.

**Comment by digitalcraftsman** [comment]
Are there still any issues with this pull requests? It would reduce a bit of overhead for us if we can point new contributors to this guide.

**Comment by anthonyfok** [comment]
Looks good to me!  +1 for merging this pull request.

**Pull request merged**